libopenage/renderer/opengl/renderer.cpp

```diff
@@ -198,6 +227,20 @@ void GlRenderer::render(const std::shared_ptr<RenderPass> &pass) {
 			glClear(GL_DEPTH_BUFFER_BIT);
 		}
 
+		switch (layer.stencil_state) {
+		case StencilState::WRITE_STENCIL_MASK:
+			setupStencilWrite();
+			break;
+
+		case StencilState::USE_STENCIL_TEST:
+			setupStencilTest();
+			break;
+
+		case StencilState::DISABLE_STENCIL:
+			disableStencilTest();
+			break;
+		}
```

Comment:
Doing stencil testing per layer is a bit overkill for now IMO, but it's a good idea :)

We have to make sure that the draw order is correct and should make performance tests for this.